File: tests/fixtures/sample.cpp
// tests/fixtures/sample.cpp
// Fixture minimo para testear c_parser.py
// Cubre: includes, defines, clase con herencia, constructor,
//        metodos virtuales, enum class, funciones sueltas, variables globales.

#include <iostream>
#include <vector>
#include <memory>
#include "sensor.h"

#define VERSION_MAYOR 2
#define MAX_BUFFER 1024
#define NOMBRE_APP "CodeMap"

const int TIMEOUT_MS = 5000;
static int instancias = 0;

enum class EstadoSensor { Activo, Inactivo, Error };

class Sensor {
public:
    std::string id;
    double ultimo_valor;

    Sensor(const std::string& id);
    virtual ~Sensor() = default;

    virtual double leer();
    virtual std::string toString() const;
    bool estaActivo() const;

protected:
    EstadoSensor estado;

private:
    void _inicializar();
};

Sensor::Sensor(const std::string& id) : id(id), estado(EstadoSensor::Activo) {
    ultimo_valor = 0.0;
    instancias++;
    _inicializar();
}

double Sensor::leer() {
    return ultimo_valor;
}

std::string Sensor::toString() const {
    return "Sensor(" + id + ")";
}

bool Sensor::estaActivo() const {
    return estado == EstadoSensor::Activo;
}

void Sensor::_inicializar() {
    ultimo_valor = -1.0;
}

class SensorADE9000 : public Sensor {
public:
    int canal;

    SensorADE9000(const std::string& id, int canal);
    double leer() override;

private:
    std::vector<double> _buffer;
};

SensorADE9000::SensorADE9000(const std::string& id, int canal)
    : Sensor(id), canal(canal) {
    _buffer.reserve(MAX_BUFFER);
}

double SensorADE9000::leer() {
    return Sensor::leer() * 1000.0;
}

double promedio(const std::vector<double>& valores) {
    if (valores.empty()) return 0.0;
    double suma = 0.0;
    for (const auto& v : valores) suma += v;
    return suma / valores.size();
}

std::string formatear(const Sensor& s) {
    return s.toString();
}

File: tests/fixtures/sample.dart
// tests/fixtures/sample.dart
// Fixture minimo para testear dart_parser.py
// Cubre: imports, clase con herencia, mixin, enum, constructor,
//        metodos async, atributos, funciones sueltas, variables de modulo.

import 'package:flutter/material.dart';
import 'dart:async' as async show Future, Stream;
import 'package:http/http.dart' show get, post;

const String VERSION = '1.0.0';
final int MAX_REINTENTOS = 3;

enum EstadoConexion { conectado, desconectado, error }

mixin LoggableMixin {
  void registrar(String mensaje) {
    print(mensaje);
  }
}

class Sensor extends ChangeNotifier with LoggableMixin {
  final String id;
  late double _ultimo_valor;
  static int _instancias = 0;

  Sensor(this.id) {
    _instancias++;
    _ultimo_valor = 0.0;
  }

  Sensor.simulado(String id) : this(id) {
    _ultimo_valor = -1.0;
  }

  double get ultimoValor => _ultimo_valor;

  Future<double> leer() async {
    final resultado = await _fetch();
    notifyListeners();
    return resultado;
  }

  Future<double> _fetch() async {
    return 42.0;
  }

  @override
  String toString() => 'Sensor($id)';
}

class SensorTemperatura extends Sensor {
  final String unidad;

  SensorTemperatura(String id, {this.unidad = 'C'}) : super(id);

  @override
  Future<double> leer() async {
    final raw = await super.leer();
    return unidad == 'F' ? raw * 9 / 5 + 32 : raw;
  }
}

Future<void> inicializar(String ruta) async {
  await Future.delayed(Duration(seconds: 1));
  print('Inicializado: $ruta');
}

Map<String, dynamic> parsearConfig(String texto) {
  return {};
}